cdap-client/src/main/java/co/cask/cdap/client/AuthorizationClient.java

```diff
@@ -223,18 +219,8 @@ private HttpResponse doExecuteRequest(HttpRequest request, int... additionalAllo
     throws IOException, UnauthenticatedException, FeatureDisabledException, UnauthorizedException {
     int[] allowedErrorCodes = new int[additionalAllowedErrorCodes.length + 2];
     System.arraycopy(additionalAllowedErrorCodes, 0, allowedErrorCodes, 0, additionalAllowedErrorCodes.length);
-    allowedErrorCodes[additionalAllowedErrorCodes.length] = HttpURLConnection.HTTP_FORBIDDEN;
-    allowedErrorCodes[additionalAllowedErrorCodes.length + 1] = HttpURLConnection.HTTP_NOT_IMPLEMENTED;
+    allowedErrorCodes[additionalAllowedErrorCodes.length] = HttpURLConnection.HTTP_NOT_IMPLEMENTED;
     HttpResponse response = restClient.execute(request, config.getAccessToken(), allowedErrorCodes);
-    if (HttpURLConnection.HTTP_FORBIDDEN == response.getResponseCode()) {
```

Comment:
Because RESTClient was not handling FORBIDDEN, the handling was done here. To be frank, all this code wasn't really needed. I think we may have been ok with just throwing new UnauthorizedException(request.getBody) here. However, now that RESTClient handles FORBIDDEN, this is not needed anymore.